src/AirshipBindings.Portable/Analytics/CustomEvent.cs

```diff
@@ -0,0 +1,161 @@
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="UrbanAirship.CustomEvent"/> class.
+		/// </summary>
+		public CustomEvent()
+		{
+			this.propertyList = new List<Property>();
+		}
+
+		/// <summary>
+		/// Gets or sets the name of the event.
+		/// </summary>
+		/// <value>The name of the event.</value>
+		public string EventName
+		{
+			get { return eventName; }
+			set { eventName = value; }
+		}
+
+		/// <summary>
+		/// Gets or sets the event value.
+		/// </summary>
+		/// <value>The event value.</value>
+		public double? EventValue
+		{
+			get { return Double.Parse(eventValue); }
+			set { eventValue = value.ToString(); }
+		}
+
+		/// <summary>
+		/// Gets or sets the transaction identifier.
+		/// </summary>
+		/// <value>The transaction identifier.</value>
+		public string TransactionId
+		{
+			get { return transactionId; }
+			set { transactionId = value; }
+		}
+
+		/// <summary>
+		/// Gets or sets interaction type.
+		/// </summary>
+		/// <value>The interaction type.</value>
+		public string InteractionType
+		{
+			get { return interactionType; }
+			set { interactionType = value; }
+		}
+
+		/// <summary>
+		/// Gets or sets the interaction identifier.
+		/// </summary>
+		/// <value>The interaction identifier.</value>
+		public string InteractionId
+		{
+			get { return interactionId; }
+			set { interactionId = value; }
+		}
+
+		internal List<Property> PropertyList
+		{
+			get { return propertyList; }
+		}
+
+		/// <summary>
+		/// Adds a string property.
+		/// </summary>
+		/// <param name="name">The property name.</param>
+		/// <param name="value">The property value.</param>
+		public void AddProperty(string name, string value)
+		{
+			this.propertyList.Add(new Property("s", name, value));
+		}
+
+		/// <summary>
+		/// Adds a double property.
+		/// </summary>
+		/// <param name="name">The property name.</param>
+		/// <param name="value">The property value.</param>
+		public void AddProperty(string name, double value)
+		{
+			this.propertyList.Add(new Property("d", name, value));
+		}
+
+		/// <summary>
+		/// Adds a bool property.
+		/// </summary>
+		/// <param name="name">The property name.</param>
+		/// <param name="value">The property value.</param>
+		public void AddProperty(string name, bool value)
+		{
+			this.propertyList.Add(new Property("b", name, value));
+		}
+
+		/// <summary>
+		/// Adds a string array property.
+		/// </summary>
+		/// <param name="name">The property name.</param>
+		/// <param name="value">The property value.</param>
+		public void AddProperty(string name, ICollection<string> value)
+		{
+			this.propertyList.Add(new Property("sa", name, value.ToArray()));
+		}
+
+		internal class Property
+		{
+			public string type;
+			public string name;
+			public string stringValue;
+			public double doubleValue;
+			public bool boolValue;
+			public string[] stringArrayValue;
+
+			public Property(string type, string name, System.Object value)
```

Comment:
Can we do something with inheritance or generics instead of sending a string type hint?